src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
+            {
+                ApplyProperties(changedObj, originalObj, propertyName);
+            }
+
+            foreach (string propertyName in changedObj.GetUnchangedPropertyNames())
+            {
+                ApplyProperties(changedObj, originalObj, propertyName);
+            }
+        }
+
+        private void ApplyProperties(EdmStructuredObject changedObj, EdmStructuredObject originalObj, string propertyName)
+        {
+            object value;
+            if (changedObj.TryGetPropertyValue(propertyName, out value))
+            {
+                EdmChangedObjectCollection changedColl = value as EdmChangedObjectCollection;
+                if (changedColl != null)
+                {
+                    TypelessPatchMethodHandler patchHandler = PatchHandler.GetNestedPatchHandler(originalObj, propertyName);
+                    if (patchHandler != null)
+                    {
+                        changedColl.Patch(patchHandler);
+                    }
+                    else
+                    {
+                        object obj;
+                        originalObj.TryGetPropertyValue(propertyName, out obj);
+
+                        ICollection<EdmStructuredObject> edmColl = obj as ICollection<EdmStructuredObject>;
+
+                        changedColl.Patch(edmColl);
+                    }
+                }
+                else
+                {
+                    //call patchitem if its single structuredobj
+                    EdmStructuredObject structuredObj = value as EdmStructuredObject;
+
+                    if (structuredObj != null)
+                    {
+                        object obj;
+                        originalObj.TryGetPropertyValue(propertyName, out obj);
+
+                        EdmStructuredObject origStructuredObj = obj as EdmStructuredObject;
+
+                        if(origStructuredObj == null)
+                        {
+                            if(structuredObj is EdmComplexObject)
+                            {
+                                origStructuredObj = new EdmComplexObject(structuredObj.ActualEdmType as IEdmComplexType);
+                            }
+                            else
+                            {
+                                origStructuredObj = new EdmEntityObject(structuredObj.ActualEdmType as IEdmEntityType);
+                            }
+
+                            originalObj.TrySetPropertyValue(propertyName, origStructuredObj);
+                        }
+
+                        PatchItem(structuredObj, origStructuredObj);                        
+                    }
+                    else
+                    {
+                        originalObj.TrySetPropertyValue(propertyName, value);
+                    }
+                }
+            }
+        }
+
+        private IEdmChangedObject HandleFailedOperation(EdmEntityObject changedObj, DataModificationOperationKind operation, EdmStructuredObject originalObj, 
```

Comment:
>EdmEntityObject [](start = 56, length = 15)

Ideally, this would take an IEdmChangedObject, to support other implementations and mocking.  That would require that any operations (i.e.,e getting the transient and persistent annotation containers) were obtained through IEdmChangedObject, or that functionality was conditional upon a successful cast.

Reply:
those operations are not available and annotation in iedmchanged object, right